spring-cloud-kubernetes-client-config/src/main/java/org/springframework/cloud/kubernetes/client/config/KubernetesClientSecretsPropertySource.java

```diff
@@ -38,49 +40,32 @@
 
 	private static final Log LOG = LogFactory.getLog(KubernetesClientSecretsPropertySource.class);
 
-	private CoreV1Api coreV1Api;
-
 	public KubernetesClientSecretsPropertySource(CoreV1Api coreV1Api, String name, String namespace,
-			Environment environment, Map<String, String> labels, boolean failFast) {
-		super(getSourceName(name, namespace), getSourceData(coreV1Api, environment, name, namespace, labels, failFast));
+			Map<String, String> labels, boolean failFast) {
+		super(getSourceName(name, getApplicationNamespace(namespace, "Secret", null)),
+				getSourceData(coreV1Api, name, getApplicationNamespace(namespace, "Secret", null), labels, failFast));
 
 	}
 
-	private static Map<String, Object> getSourceData(CoreV1Api api, Environment env, String name, String namespace,
+	private static Map<String, Object> getSourceData(CoreV1Api api, String name, String namespace,
 			Map<String, String> labels, boolean failFast) {
 		Map<String, Object> result = new HashMap<>();
 
 		LOG.info("Loading Secret with name '" + name + "' or with labels [" + labels + "] in namespace '" + namespace
 				+ "'");
 		try {
-			// Read for secrets api (named)
 			if (StringUtils.hasText(name)) {
 				Optional<V1Secret> secret;
-				if (!StringUtils.hasText(namespace)) {
```

Comment:
since we know the namespace for sure here, we can get rid of calls `api.listSecretForAllNamespaces...` This will also close [this defect](https://github.com/spring-cloud/spring-cloud-kubernetes/issues/861)